proxy-node-test/src/main/java/uk/gov/ida/notification/helpers/EidasAuthnRequestBuilder.java

```diff
@@ -48,7 +51,7 @@ public AuthnRequest build() throws TransformerException, SignatureException, Mar
         if (this.signingCredential != null) {
             SignatureBuilder signatureBuilder = SignatureBuilder
                     .aSignature()
-                    .withSignatureAlgorithm(new SignatureRSASHA256())
+                    .withSignatureAlgorithm(Optional.ofNullable(signatureAlgorithm).orElse(new SignatureRSASHA256()))
```

Comment:
Looks good - and keeps the default behaviour. Should it? Or at some point do we want this to fail if the signature algorithm isn't explicitly specified?

Reply:
We don't have any logic in the ESP to fail validation if the signing algo is outside the intersection of key length & required algo. 
Was going to talk to @CharlesIC about this